rust/quic/src/quic.rs

```diff
@@ -0,0 +1,263 @@
+    use nom::IResult;
+
+static QUIC_REQ1: &'static [u8] = &[
+    0x00, 0x02, 0xb3, 0xb1, 0x7f, 0xa8, 0x89, 0x5c, 
+    0x77, 0xeb, 0xcb, 0x8d, 0x90, 0xe5, 0x51, 0xc9, 
+    0xcd, 0x13, 0xdb, 0xbf, 0x45, 0x65, 0x11, 0xf6, 
+    0xfc, 0xf9, 0xbf, 0x32, 0x6a, 0xb8
+];
+
+#[test]
+fn test_quic_packet1() {
+    let empty = &b""[..];
+    let bytes = QUIC_REQ1;
+    let session_id = [0x02];
+    let expected = IResult::Done(empty,QuicPacket{
+        flags_has_version:false,
+        flags_has_reset:false,
+        conn_fld_len:0,
+        version:None,
+        seq_fld_len:1,
+        connection_id:None,
+        seq_id:&session_id,
+        payload:&bytes[2..],
+    });
+    let res = parse_quic(&bytes);
+    println!("{:?}",res);
+    assert_eq!(res, expected);
+}
+
+static QUIC_REQ2: &'static [u8] = &[
+    0x0c, 0xe3, 0x67, 0x86, 0xee, 0x6e, 0xc8, 0x56, 
+    0x62, 0x07, 0xc0, 0xcf, 0x74, 0x6e, 0xef, 0xe1, 
+    0xd1, 0x64, 0x76, 0xa4, 0xfd, 0xec, 0x0b, 0x74, 
+    0x3b, 0xa0, 0x13, 0xef, 0xd9, 0x98, 0xc7, 0xbf, 
+    0x04, 0xda, 0x47, 0x35, 0xf3, 0x7e
+];
+
+#[test]
+fn test_quic_packet2() {
+    let empty = &b""[..];
+    let bytes = QUIC_REQ2;
+    let session_id = [0x07];
+    let expected = IResult::Done(empty,QuicPacket{
+        flags_has_version:false,
+        flags_has_reset:false,
+        conn_fld_len:8,
+        version:None,
+        seq_fld_len:1,
+        connection_id:Some(&bytes[1..9]),
+        seq_id:&session_id,
+        payload:&bytes[10..],
+    });
+    let res = parse_quic(&bytes);
+    assert_eq!(res, expected);
+}
+
+static QUIC_REQ3: &'static [u8] = &[
+    0x04, 0x5e, 0x78
+];
+
+#[test]
+fn test_quic_packet3() {
+    let empty = &b""[..];
+    let bytes = QUIC_REQ3;
+    let session_id = [0x78];
+    let expected = IResult::Done(empty,QuicPacket{
+        flags_has_version:false,
+        flags_has_reset:false,
+        conn_fld_len:1,
+        version:None,
+        seq_fld_len:1,
+        connection_id:Some(&bytes[1..2]),
+        seq_id:&session_id,
+        payload:&[],
+    });
+    let res = parse_quic(&bytes);
+    println!("{:?}",res);
+    assert_eq!(res, expected);
+}
+
+static QUIC_REQ4: &'static [u8] = &[
+    0x0d, 0xe3, 0x67, 0x86, 0xee, 0x6e, 0xc8, 0x56, 
+    0x62, 0x51, 0x30, 0x33, 0x35, 0x01, 0x19, 0xad, 
+    0x47, 0x62, 0x6b, 0xf0, 0x9a, 0x19, 0x45, 0xc2, 
+    0x58, 0xa8, 0xa0, 0x01, // truncated the payload
+];
+
+#[test]
+fn test_quic_packet4() {
+    let empty = &b""[..];
+    let bytes = QUIC_REQ4;
+    let session_id = [0x01];
+    let version = [0x51, 0x30, 0x33, 0x35]; // Q035
+    let expected = IResult::Done(empty,QuicPacket{
+        flags_has_version:true,
+        flags_has_reset:false,
+        conn_fld_len:8,
+        version:Some(&version),
+        seq_fld_len:1,
+        connection_id:Some(&bytes[1..9]),
+        seq_id:&session_id,
+        payload:&bytes[14..],
+    });
+    let res = parse_quic(&bytes);
+    println!("{:?}",res);
+    assert_eq!(res, expected);
+}
+
+}
```

Comment:
Globally, except for the debugging code, the parser and the unit tests look good to me